src/script/filter.js

```diff
@@ -20,31 +23,51 @@ export class Filter {
     */
 
     filter () {
-        const renderer = new Renderer();
+        const renderer = new Renderer(this.data);
+        const filterByTag = new FilterByTag();
+        const filterByType = new FilterByType();
+        const filerByRating = new FilterByRating();
         const filteredList = { challenges: [],};
 
-        this.data.challenges.forEach(challenge => {
+        // Checks all filter elements if they are empty, if they all are empty (returning false) load all challenges
+        if ( 
+            filterByTag.checkDOM() ||
+            filterByType.checkDOM() ||
+            filerByRating.checkDOM()
+            ){
+            
 
-            /* if( filterByTag(challenge) &&
-                filterByText(challenge) && 
-                filterByType(challenge) && 
-                filterByType(challange) )  */
-            if (filterByTag(challenge)) {
 
-                filteredList.challenges.push(challenge);    
-            }
-        })
+            this.data.challenges.forEach(challenge => {
+                // Checks for each filter if the challenge matches the filter (returns true) or if all the DOM elements for that filter is empty
+                if (
+                    (filterByTag.filter(challenge) || !filterByTag.checkDOM()) &&
+                    (filterByType.filter(challenge) || !filterByType.checkDOM()) &&
+                    (filerByRating.filter(challenge) || !filerByRating.checkDOM())
+                    ) {
 
+                    filteredList.challenges.push(challenge);    
+                }
+            })
 
-        if (filteredList.challenges.length > 0 ){
+            // If any challenge got through the filter, filteredList.length will have at least one challenge and will render that challenge
+            if (filteredList.challenges.length > 0 ){
 
-            renderer.renderRooms(filteredList)
+                renderer.renderRooms(filteredList);
```

Comment:
Nice delegating of responsibility to a separate class! I have some thoughts on the specifics of how this is done though. See my comments below.